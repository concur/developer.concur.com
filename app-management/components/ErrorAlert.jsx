import React, { PropTypes } from 'react';

const ErrorAlert = ({ error }) => (
  error ? <div className="alert alert-danger">{error}</div> : null
);

ErrorAlert.propTypes = {
  error: PropTypes.string,
};

export default ErrorAlert;
