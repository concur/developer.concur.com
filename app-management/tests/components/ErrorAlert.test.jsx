import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ErrorAlert from '../../components/ErrorAlert';

describe('<ErrorAlert />', () => {
  it('should render if error is present', () => {
    const alert = shallow(<ErrorAlert error="Fetch failed" />);

    expect(shallowToJson(alert)).toMatchSnapshot();
  });

  it('should display nothing if there is no error', () => {
    const alert = shallow(<ErrorAlert error="" />);

    expect(shallowToJson(alert)).toMatchSnapshot();
  });
});
