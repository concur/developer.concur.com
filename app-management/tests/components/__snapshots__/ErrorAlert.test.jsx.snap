exports[`<ErrorAlert /> should display nothing if there is no error 1`] = `null`;

exports[`<ErrorAlert /> should render if error is present 1`] = `
<div
  className="alert alert-danger">
  Fetch failed
</div>
`;
